include/libhal-arm-mcu/stm32f1/pin.hpp

```diff
@@ -57,4 +57,14 @@ enum class can_pins : std::uint8_t
   pb9_pb8 = 0b10,
   pd0_pd1 = 0b11,
 };
+/**
+ * @brief Default pins for pwm tim1 channels
+ */
+enum class pwm_pins : std::uint8_t
+{
+  pa8 = 0b00,
+  pa9 = 0b01,
+  pa10 = 0b10,
+  pa11 = 0b11,
+};
```

Comment:
Do we only support timer 1? What about the other timers?

Reply:
There are other timers, however the user manual describes the pwm only for Timer 1 and Timer 8. The other timers might be able to support pwm but with slightly different registers. (I'm not sure I haven't looked into it too deeply). And the stm32f108x datasheet did not have any pins for Timer 8, so thats why I haven't added that in. There is an option to use different pins for the various timer1 channels if we choose to remap them, but I haven't implemented that. I can add that in if requried.

Reply:
Hmmm, okay. Lets rename this to `timer1_pwm` then, since this seems to be constrained to timer 1. 
I'm thinking about what it would take to support different pins. After construction you don't need the pin information anymore and thus you can drop it. So if we made a new constructor, we could change the parameter to handle a wider variety of pins. This is important, because once you define the size of this class, a change in the object size is an ABI break would can cause undefined behavior in the program.